site/en/docs/extensions/mv3/storage-and-cookies/index.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Storage and Cookies"
+seoTitle: "Chrome Extensions storage and cookies"
+date: 2023-08-01
+description: Overview of how web storage APIs and cookies work in extensions
+---
+
+Extensions can store cookies and access web storage APIs similar to a normal website. However, in
+some cases these behave differently in extensions.
+
+This article does not cover extension specific APIs. For more on these, see
+[chrome.storage][chrome-storage-api].
+
+## Storage {: #storage }
+
+### Persistence {: #storage-persistence }
+
+Extension storage is always persistent. For example, when a user uses the "Clear Browsing Data"
```

Comment:
I'd like to check this with Devlin. The code seems to make it very clear this is the case ([source](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/extensions/extension_special_storage_policy.cc;l=123;drc=4ddfd6785f21a7bce85e7adccbfc6edcb7e73ef1)) but https://bugs.chromium.org/p/chromium/issues/detail?id=1209236 makes it sound like it isn't. So I don't want to make any guarantees and then realise I missed something.

Reply:
On a second reading of the code, I think memory pressure evictions can occur for extensions without the unlimitedStorage permission.

Reply:
Should you also mention how web storage works with content scripts?

Reply:
I've added some information which hopefully helps here. This is the same for popups and the side panel, but content scripts access storage from the website's origin, so it doesn't really make sense in the same context.